### DISCUSSION TITLE
subscribe to a record not working

### DISCUSSION THREAD

**Discussion by islamzaoui** [pocketbase/pocketbase#4380] [Q&A] [answered]
Hello guys
I have a problem with subscribing to a current user record in sveltekit
```svelte
<script lang="ts">
	import { pb } from '$lib/pocketbase';
	import { enhance } from '$app/forms';
	import { onDestroy, onMount } from 'svelte';
	import type { UnsubscribeFunc } from 'pocketbase';

	export let data;

	let requested = false;
	let unsub: UnsubscribeFunc;

	onMount(async () => {
		if (data.user)
			unsub = await pb.collection('users').subscribe(data.user.id, ({ record }) => {
				console.log(record);
				if (record.verified) window.location.href = '/app';
			});
	});

	onDestroy(async () => {
		if (unsub) await unsub();
	});
</script>

<div class="m-auto flex h-screen flex-col items-center justify-center gap-10 border">
	<span class="text-2xl font-bold">You need to verify your account to continue</span>
	<form method="POST" use:enhance>
		<button type="submit" class="rounded-xl border p-4" on:click={() => (requested = true)}
			>Verify my Email</button
		>
	</form>
	{#if requested}
		<span class="text-2xl">This page will refresh when your email is verified</span>
	{/if}
</div>
```

this working fine when view and list rules are empty string 
but when I set them to `@request.auth.id = id` it's wont work any more

can you help me fix that

pocketbase version 0.21.3

**Comment by ganigeorgiev** [comment]
> this working fine when view and list rules are empty string
but when I set them to @request.auth.id = id it's wont work any more

It works when it is an empty string because in this case the actions will be allowed for everyone.

It may not work with `@request.auth.id = id` because you are probably not authenticated.
Note that the JS SDK doesn't know what `data.user` is and your `subscribe()` call will be authenticated only if `pb.authStore` has valid  token.

**Reply by islamzaoui** [reply, marked as answer]
yeah my bad, the client authStore wasn't linked with server authStore 
I just created hooks.client.ts
```ts
import { pb } from '$lib/pocketbase'

pb.authStore.loadFromCookie(document.cookie)
pb.authStore.onChange(() => {
    document.cookie = pb.authStore.exportToCookie({ httpOnly: false })
}, true)
```
